kore/src/SMT/SimpleSMT.hs

```diff
@@ -281,48 +285,37 @@ newSolver exe opts logger = do
 
     return solver
 
-wrapScope :: [Logger.Scope] -> Logger.SomeEntry -> Logger.SomeEntry
-wrapScope scopes entry =
-    foldr (\s e -> Logger.SomeEntry $ Logger.WithScope e s) entry scopes
-
-logWith
+logMessageWith
     :: GHC.HasCallStack
     => Logger.Severity
     -> Solver
-    -> [Logger.Scope]
     -> Text
     -> IO ()
-logWith severity Solver { logger } scope a =
+logMessageWith severity Solver { logger } a =
     logger Colog.<& message
   where
     message =
-        wrapScope scope
-            $ Logger.SomeEntry
-                $ Logger.WithScope
-                    { entry =
-                        Logger.SomeEntry
-                            $ Logger.LogMessage a severity callStack
-                    , scope = Logger.Scope "SimpleSMT"
-                    }
+        Logger.SomeEntry
+        $ Logger.LogMessage a severity callStack
 
-debug :: GHC.HasCallStack => Solver -> [Logger.Scope] -> Text -> IO ()
-debug = logWith Logger.Debug
+debug :: GHC.HasCallStack => Solver -> Text -> IO ()
+debug = logMessageWith Logger.Debug
 
-warn :: GHC.HasCallStack => Solver -> [Logger.Scope] -> Text -> IO ()
-warn = logWith Logger.Warning
+warn :: GHC.HasCallStack => Solver -> Text -> IO ()
+warn = logMessageWith Logger.Warning
 
 send :: Solver -> SExpr -> IO ()
-send solver@Solver { hIn } command' = do
-    debug solver ["send"] (buildText command')
+send Solver { hIn, logger } command' = do
+    logDebugSolverSendWith logger (buildText command')
```

Comment:
I would prefer if `logDebugSolverSendWith` took an `SExpr` as its second argument. We do not need to render the `SExpr` to text unless the log entry is displayed.